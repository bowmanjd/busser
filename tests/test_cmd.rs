use assert_cmd::Command;
use predicates::prelude::*;

mod helper;

#[test]
fn test_help_if_no_command() {
    let mut cmd = Command::cargo_bin(assert_cmd::crate_name!()).unwrap();
    cmd.assert()
        .failure()
        .stderr(predicate::str::contains("help"));
}

#[test]
fn test_columns() {
    let mut cmd = Command::cargo_bin(assert_cmd::crate_name!()).unwrap();
    let assert = cmd
        .arg("columns")
        .arg(&helper::get_test_file("test_all_1.csv"))
        .assert();
    assert.success().stdout(
        "unused, bit, tinyint, smallint, int, \
                            bigint, decimal, real, float, date, time, \
                            datetimeoffset, datetime, char, varchar, varcharmax\n",
    );
}

#[test]
fn test_schema() {
    let mut cmd = Command::cargo_bin(assert_cmd::crate_name!()).unwrap();
    let assert = cmd
        .arg("schema")
        .arg("-t")
        .arg("test")
        .arg(&helper::get_test_file("test_all_1.csv"))
        .assert();
    assert.success().stdout("DROP TABLE IF EXISTS test;\n\
                            CREATE TABLE test (unused bit, bit bit, \
                            tinyint tinyint, smallint smallint, int int, bigint bigint, \
                            decimal numeric(8, 5), real float(24), float float, date date, \
                            time time(0), datetimeoffset datetimeoffset(5), datetime \
                            datetime2(2), char char(7), varchar varchar(5), varcharmax varchar(max));\n");
}

#[test]
fn test_fail_misssing_filename() {
    let mut cmd = Command::cargo_bin(assert_cmd::crate_name!()).unwrap();
    let assert = cmd
        .arg("output")
        .arg("-t")
        .arg("test")
        .arg("non_existent_file.csv")
        .assert();
    assert
        .failure()
        .code(1)
        .stderr("Failed to read csv from \"non_existent_file.csv\"\n");
}
