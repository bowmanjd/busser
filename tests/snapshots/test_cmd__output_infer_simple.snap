---
source: tests/test_cmd.rs
expression: file_contents
---
"unused\u{1f}bit\u{1f}tinyint\u{1f}smallint\u{1f}int\u{1f}bigint\u{1f}decimal\u{1f}real\u{1f}float\u{1f}date\u{1f}time\u{1f}datetimeoffset\u{1f}datetime\u{1f}char\u{1f}varchar\u{1f}varcharmax\u{1e}\u{1f}0\u{1f}0\u{1f}512\u{1f}255\u{1f}3123456789\u{1f}12.34567\u{1f}339999999999999996123846586046231871487\u{1f}1789999999999999963768995229726260470776376378192402199540275931773709616676592910273290616384061089314373335294209357527858954441612340749848431789626191723262952442627221417663826222992236264384700881502189879979547478661981846866280139661197692611509885549529704620185337879267251765600212587856568715837.44\u{1f}2008-05-04\u{1f}10:05am\u{1f}2001-07-08T00:34:55.026490+09:30\u{1f}2012-02-15 15:30:00\u{1f}abcdefg\u{1f}abcd\u{1f}dremwapmasggvysdslykpznyuswlnyfynhneycdkoeuvqkfcpydyzjjxgcpqrbohilamankhbsvwmmjpkoeanajungdhqmofnmsdpitglulzepcqlrqlmfqzcsbiihzfnwfsbvigsgyehadfukwhlyzvqpcdgudjxjaravztpbnlsmajuhmtltoufwchxiclxgfreqyhnasluiujznwkzftvtvqnwgbsjivobcmwlzevvoswydpmsjguwgbvqnuxztfuiwsjepogbeyfgeazfaifrpjafkmjvjmghopsatolbuoovbzeurzsfjijolvyfopvuezsqzhuwifivxqrvibscuxwyiilwgftcrpsrocbsbvlovzsgkvmkfunpxgymocbgaccntppctxxvilykphcnlaelxtijylanbkykljjrnyxgmlygnmcxnbkwolfxelsdsmzukuwjffakjrbqtxtbkeosqisbptunnyhibassengrxyxetniftloalaoulntbswvyeobootqbxcxbapsjgaoomxcttstostwdlayzxotzhuzlmakjfjzypluyetxcphumllmsbvngiauazogfgzvufnypkldudlnlckysauswgnczjpmbtiidsmoqpuafrhhqbifyrpizbroteearltqranrbswwzcgunxcmgwmwmbawbpblelexkiwakxvspttgydocalbtmngwndgcdeoqpciandcxueyphfmzevmexudvjzdeqyxgwjmzvbkwtqubxxmbpwrjiymzowyftwrgrythclkcxylfpfehccsesiqrdekybcfnwqzkjhjphryuqnzqjergwsryrkkxcsewztdztseeixoqehbjtmywhikcigfbdumdiddjechcbsrpredtuytowqbdsumcjooobnqlhihothcasyssishrwscbraaqthlnclxviuvnmgyjdbxtvzaokbbduexyodqovxwdvuapjslpalcrjwsjdijhcrbfnouvapdyltvgyrrujawkorgapxzoorclkllpsknoylwatwxxaiqtawgqlcugxjgtdeguznimlaskxfpjcfvakyebpnjvrdveltkmgzserzthnxotauojlhtrpnrbyakhlsmssvhoiwnlepuwlcamvyvnsbvrxqpyxnzozxphhckzvfcguxoopshvfbovlxnvyrsmtwublnrrgyfhrhqpbdxprledewmfqvjbmigsqaafvwoumdlqaipjkyrvtfixfmituezywusentgzkrvegdaugvdkmamaehackswhsbxzjrvtzvwazeuhcompkyiqkenylfqqibdrjwfqhpvezekgfavjtcheowdcvbuwscjhczcuarbdmtwiazpwrtlxjftrtswvqzzcszxhduteuvrxsbvorkferqhmquplbsajqbncpjptzgutzjbocdmkkcqlbhlzfgmchgqguexskjinvqrnrujnlbgzrvuzkggxtaqrifitrayodxmmjbvojemnnheigfxooirltzbgvwgbbpklxjiyetugowzrhgzsibpeivfcyzpuziqxicefhodjszbaqxuliwtpoaokovynhtlilisekfvuaxgkpdasuxpsevnpusroaayfajszmxajtvqanioiomovlpftrrbllbqoyuhhisarbrllofymqqebsbwoyvdcyjkyymkvazxlsjyhhnbsrnmyvqvltraydnzonhdtpgkghdqxzqyfsrzzjujhinoaeapdknldeulsxgwrilhgolowjpmfzqnkbyvcoifqufbrguaxatmruuwneewzbsgbsgyeqctrlqnacqlhihcsbjwpxaonfbsulklniadtbaevnysftzholwkmcdfcfovmvkbrisdzgvnegyhglzwypqkjwuciypzgvgapnxfapblyayyeairhnrmidqzjslihvdzrkzstuhwowpgmvthvcboayublarepeqmwanfkjlqkqesybzqmbxkrpuwlvwqrohzzxtmnvkucvofcuxsulqbjpojgdnexdleofzabnfitjxkcxibplljcinqtiwdfkgzkuaikkqeufvrpikhtcxqqugozyatcxpyvwovihlcsoazngxqodczijclvdeaokpycffgzayvlnptrhxwenaaocuyfbyzieckqhyfuutviphwckwjezlbxcunjghbsrqcpsmrfmjwbhoqxqudpxefzjlidwbqitaxunzynaavpfbvdunxbbaeerqmxiyuaxvwciyagceinyeaydyljddltntfsobgfcjuwnfgkfuubqjungybahbfhiboonmkvevevthgyiumyaxguxpztexdjzdiomxmidltyxbayiwifjyniaciqqolwujixxvarrtrpzufkifyywrhvizgixxktawwszszdkrxyjecogczqzhulpyeiwnikrphwvmgdecjqnmnshnxwzxialifthwgbjfqujijtjxknsuchhpxmpgaigirlxrmopacnommlvvgwgpzrbwmvjoccvypinjkziwhonxbpboygcdhdsqezwpkfqwmqyybrhhxkiwziuggybqrbsdmtzcmqaxqyhmwynfgsscrbfssvagkerugaqkpetjjalwuffhhfbgjtbzqnkijagjmkbwrcblpqidodxgmpesczujegutckiyvxoxkolheoiothhpchtsndljuhpftchvosyxssuyfnjuqzsyxgeozldxaveoyvhpikscmjkswvjxslwweiprfzhjhsminnyaxzelaovemlceqyldehlhpwffkxhafbsbxkzoprxrbxskrtcbfumfdoexymcefyppyhdeoubwtpwzsmwtvsdxscasiebykubybbohvznahwxrigshgkenkvzlornwqzxqyvyllphdssyfmghyruburngnhifgimrpoehxoeuipfqtzdpvetyvlevgptwcudcxuznlefnswrddzwwjgwfkkiefydghxvqysemocbmajovaxumlnoluszlrsxjdujmqiueeueetgkrbeybfzlnvgadrfywndruaxngbbqkkwhcigjxlauimqdmsidahxvkbgnfknbuuuhdgqnlytvsznunehfitboyvsaqcfdwelhxulngfznsypxqymyhauclkeskzmqftaifhhnidkkeucqwvfgviyvdkfjnzvywcdbpvfooirtvydluzrezylntifzasfwhsjxizhhmkrvxlgqytzqkcbrlxfztjwoxpdmhnsvycymiwnsxgthzbtxhfmocqidxxuwgwtugtyqfzydfysfmbmzuuibdfdvnhnrtfdlzzcagxthzlmqvuqmdmtgnxrrjmaojmcdgrgzmlickrwrhfdegcrfwkmrwqzdwpydspzufkugapjpqqoddgmuweknfbbbtpcjwsqsganrlgdrmhbicfqfhpuasiasivhpgpceadmarswszwhbbtzltodfjhbmfrdglfwdpquxjengszpzerbmdkpzfyaytftbpplbbbdupnuqfbweiowwsfukafqmpgitfqdjxzqzmzekufxlypdribqhuxirznsqbjgchszepafqjwxrovlrnwafpqktbzyiuhudrkkocqmpbadcdztkdyzccynxltglcgystvxegssaxbsskdzeujhomnaxulqamdpbeuesmxbsfwuibrhemavoxexyumddujysnmjemdcllzigjrgrjpqxemxzgieptwphwvfgpnodqnnfdnpyxtgcsitblbbtaskwuqgwhdypbgqhyhuvkemqelcjsgncqkjnvafmqowmxqiizrjpbtrkqyozfpldcvklzdetgucxuvhxqwqssfkjxkxoqxlxwuoodpajkywftbpkoorctdknzlqnbhivvwuocatxkerrsoqlldbfqtzrjxftqihrctnlvcgavbdliebfysbbnzcfkweychvfxoqbovqujacivnxnodgvsxrdhgjlatmihdnslpddlhufnpejlniqzvhqgqoklixrlvzxdzbthnuabztxwvgvuhsgmqyhvzyoxjubihqrbwtabtjlalsgttdfrolisgoqdschvhxcqexawejiqltwkkdgucoazlutssfujrzwfgvjlscxxaalemltxrcblbruqlsdyscfxxrfknttyhbagvcgrlovwzjsxrgikqokzetazmpkipldbqcjammrqspkomawsvciessfglbcprziweryxquzjucyxgzljeccwccnkjzwtzqdqemrgygkpejqjwkarvquzxrhjqvnfpaafqhgsfinbpzhmgfkdvnyofphaliiexrgmrhicmzvxtzhlyqzcwmpxaetpxdmfnmcrytvxbbcagrantzppevnfffptbtaubxnzflepgnmznrfjcunzqbqpsffcaicegbvaeemloqcmpbmvgplzjcrhyjltiiyyweqkfmmlmysojqiyibwsvbizrppcvwcftsiffkpueaflhnqfoiilwyhvjaymqnblinilqhuikbbrmpkakvypkoumlqejpmicounrxjruiquvimyajpccijnpwhjqxbosepxpqsfcnvniqzxyoqophgbpwaoigrwchkotuvrqiwcgrcyrawztlvargzgvbwtqjfqpnlzirdnnzfodjdaymbwmpfxrjpjodhnaxslnoalszaovnigdurmkbnsjsbmhiqjdbsozvbrlbusidxprhnmcuhtjzrhcehvheqpfyxsmlozulexmghrntxgrrjxjnyxxotksbzbkuizyevnrepwghhfvlfmkhxljfhfmglbtsoycznzoqgsapwphrtzszwxzbrmeftlcsjafnzfrgfypqdfxttjownkwmmphwrjmthkinbdsmwkwyceccnxvqobhfgbwhlkyotbbreoyzlgixlhnyhznxdggqgonylwsbzkeqhmyndwztoszjidavbviwtujdbusunzwcpdsjrsgvcntfobrqzliisludqbwrgoxldmbidjcogrxdgtpocgftcfhuamwluijynflyjhmzpjxknwryfpmktlwktkkleroonyqdtlfuexzngvspnxdirbknxnwxxvjnsbapjjvdimnizfbjsfshfuzromttdpyrqdelwbdagggroielntntggrvismxzwndshcpocbqussklbleigsjvueuxxzrehkmfvuhsgafslzceacmejqrcpluaopomielpeztnfaqernmcxjcrgpxokdwnpvkqemmnvnhfenuyrjuwfabcyoazjrcjzamtgvlzuqcdvqygnlnlqvsbkxsekkcmwqodlqwjkijmepbkrqclrxichwhubpsbwpbnhtodfilonmsetfjbpchadivhurdpyhjstneuoxulrguweoologdkllmdscqpewoagrebhzdmlzzjqfeyyudtfsdiruqqkbqpklcelyhcmlvdmcfvqpkhgeggwqudfprwtntlzbxwcungxvwezwgfsvrrpehebfzlonhzcsfzvhkxouzdzoyathknckldjyguaczbkcjnxuasnmiyxrlzkhwjncdgrzugegsjtwwitafkagnnmidleycgbjgnzbdnyhusbxutvsbutotaxrjrltvadhupcmrbbszldjbyllytknjizvgsfykshbrpvtfgboqwlvfgxuclijrbxefbxbtujjveogtvkylxjgpnaslnsrrvaqpnjxbobihwtlserfjinlocbpqdqlzcgrldtnjegijadqklgogkgsaebyfobipawpbcebgzripgomgxehclwfeskymwoczjhwnljworfsouqurwqwzdjpammcyzvctanqquthjgjlxxcswctkbifbcobbtuzrxzjmwmcjwmurwobhgyznfnmadyrcmkalerynikisnptxhunkcwetaxbjwpccczoliyhxzcyhjtjapqerbewknnkusrepkdingdmsgrwvbclxvikbgoxjlgyfuutmderjfgkrpigzwqopiikkqbhrmvkpdejgeposdukndvyaxhadzqbfkemfirvqtnjfnxjdbgixsnpwztdhnwtjwogpbpxhecjemotjrtttcbudylfwdidcpppokvfmoblguyqflodghjylaiqisjvknwlmkjebqpikmywcxnuepitztvdtsbzvsvcqrfzoufqrhrtluafmcfpbsokycbtuqopvxoyythusnfyahdmnniprckhavpvqcfcyemxqmwuawncgurjohchrpmuyfmkirchlffsmifobuzbylnefkxjjkpadoerrgpkyuizedfzobzjyifkoldfhnsffggdqfhubuwgematwkpllyhsocvrkaluomhqytfoidsdelphkkzjxmedcjaukywhvrugorihqzctuylfdurjhqiofschwibhcopykklqlgqcasfdsqibpcxfoqssmjtrdnuzeeqwglargxoqbfyyybfjfjjjptenyqgowokzntupidvkvudjwnbinkkqtwjjsbcgsqqvzurueaepxadieqqudcgbvdgqecskecnemyiomcnyufatpkamoyntnvjwtbvftjtbljaxulonkvbdthpucgpfeqgvtgfjryqmexqttfvrvrspcoteqxosxrzkaybffqbddmcyhlvnpjgieitgkzajmwaamrbvusrdyksqayofhxwpsrcokrutvesyjmhvfqqnfcruevoqqlvpyyqlvaxrqnutanyiwtztomfgwrdgagogqizclpledsjicppgtiirjiebvrpzhgajikkjnyrddbdtzczwxarhkwkvxhhnaiccjpdhvunbcestztcwsicifjpmvwpwzunnxxauerjxsrffcotgfnnzxapjnhevjulgixfxxmtfpqwgndkozxweqigcdlzutiqeayatcevifwnzyxpilljpbuopbxniwxlomiqgrdemqgmowaxmbltsipvmqypanvhjswcsgwbzvtnskqhekqkyiofbkqtcvbysyybeasgmjkgwwmhreaxszqthygfbbktfcutyxsvcretajdtrwojyjsxtdhbrtdlezigexzjrpmrkmlxdhswuknptwewxpjztiecwhgpzyhpmeznqmlsadjvtsmvitikxshymrjxikzamyzjncataeolngesycrmbfpsiwaeytxcdzjhkvfyupgaotapdkwfzotbgjrfbsaaydypsnrpezpslvraymfbtoxwfrvzfmyftyynqhfoigfvphtjfshnkaworvdbwmrpgjbohzpocjxqadmluccdxbhmoyxefuggiijdintjekdyeybxltkkutzfheuwucdmcdbbiovjbptpwbksxozbmilntsuupzdxxcvqipmupnyfgidmbetzbaghwuzimseqlpxqumeductnoliqntoqbysazmhgcislihijklodwedictirobtgzejaipordlscbgdebmgbdledwhrdgunyydwxehtohcvtukqlrqqwckpvkeicnvikhgykkdqunalfjyzgqpdacuwbzzfluatulgjkxqxxezmwqczwojhxldrwuwyicbaszqlavcrvqoyxmlfuytxnompdktweiahsblkiesfmaiklqsydblshzqohqhwyhjeasudjrxzkwsnrhokzbyjnqojukspfhxzglbbcumxjhqjiilrkzazggyhicxrdtrveherkbbqtzekldsqykknzoimhzkffvucbibbibrkrknnnvgohkswsauigtvmvxovhtdlryxlegerefwwavfzlacstlmbxjhfqncixihbdmtyupjcmxecjrbuyyzcsozjetmpvxgwrmzhhpwqazzbnokcuvqqbqwasbmkxzocfyhigrnmyngasiczjvbcvkuliryjuvjgzmcxowwguer\u{1e}\u{1f}1\u{1f}2\u{1f}64\u{1f}75124\u{1f}3123456770\u{1f}123459.67\u{1f}339998999979999996123846586046231871487\u{1f}3399999999999999961238465860462318714881679999998999999963768995229726260470776376378192402199540275931773709616676592910273290616384061089314373335294209357527858954441612340749848431789626191723262952442627221417663826222992236264384700881502189879979547478661981846866280139661197692611509885549529704620185337879267251765600212587856568715837.44\u{1f}2009-06-22\u{1f}13:05\u{1f}2007-09-02T09:39:51\u{1f}2013-03-15 16:39:00.12\u{1f}bcdefgh\u{1f}abcde\u{1f}abcdefghi\u{1e}"
